src/rulesets/Base/array.jl

```diff
@@ -233,14 +234,9 @@ function rrule(::typeof(hcat), Xs...)
                     d > ndimsX ? 1 : (:)
                 end
             end
-            dX = if ndimsX > 0
-                # Here InplaceableThunk breaks @inferred, removed for now
-                # InplaceableThunk(dX -> dX .+= view(dY, ind...), @thunk(dY[ind...]))
-                dY[ind...]
-            else
-                # This is a hack to perhaps avoid GPU scalar indexing
-                sum(view(dY, ind...))
-            end
+            dX = @allowscalar dY[ind...]
+            # Here InplaceableThunk breaks @inferred, removed for now
```

Comment:
is this still relevant? I guess it broke inference because of the `if` statement above?

Reply:
Indeed, have now restored InplaceableThunk